### PR TITLE
Add support for Common Lisp

### PR DESCRIPTION
**Pull request by sionescu**
People who have pledged to maintain the Lisp support:

Stelian Ionescu @sionescu, Thayne McCombs @tmccombs, Luís Oliveira @luismbo and SANO Masatoshi @snmsts.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
Thanks for opening this PR. Please read https://docs.travis-ci.com/user/languages/community-supported-languages/ and follow the process described there.

**Comment by sionescu** [comment]
I'm already following the process: for step 1 we're already 6(https://github.com/orgs/lisp-travis/people). And this is step 2. If there's anything I missed, please advise.

**Comment by BanzaiMan** [comment]
The maintainers are not identified in this PR.

File: lib/travis/build/script/common_lisp.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ros -e '(format t "~A: ~A~%" (lisp-implementation-type) (lisp-implementation-version))'
+]
+        end
+
+        def script
+          if config.has_key?(:system)
+            system_keyword = ":#{config[:system]}"
+            sh.cmd %Q[\
+ros -e '(ql:quickload #{system_keyword})' \\
+    -e '(unless (asdf:test-system #{system_keyword}) \\
```
Comment (luismbo) [inline lib/travis/build/script/common_lisp.rb]: This doesn't look right. `asdf:test-system` always returns `nil`, doesn't it?

File: lib/travis/build/script/common_lisp.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        ROS_URL = "https://raw.githubusercontent.com/snmsts/roswell/release/scripts/install-for-ci.sh"
+
+        SYSTEM_MISSING = %Q[\
+No test system or test script provided.
+Please either override the script: key
+or use the system: key to specify a test ASDF system.
+]
+
+        def configure
```
Comment (luismbo) [inline lib/travis/build/script/common_lisp.rb]: Does this run on every build or is it cached somehow?

File: lib/travis/build/script/common_lisp.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          super
+          sh.cmd 'sudo apt-get update'
```
Comment (luismbo) [inline lib/travis/build/script/common_lisp.rb]: `cl-travis` needs to pass `-y` to apt-get. Is that not needed here?

File: lib/travis/build/script/common_lisp.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+          sh.cmd 'sudo apt-get install libc6:i386 libc6-dev libc6-dev-i386 libffi-dev libffi-dev:i386'
```
Comment (luismbo) [inline lib/travis/build/script/common_lisp.rb]: https://github.com/luismbo/cl-travis/blob/master/install.sh#L31

Don't we need `dpkg --add-architecture` here?

File: lib/travis/build/script/common_lisp.rb (hunk 2 above)
Comment (sionescu) [inline lib/travis/build/script/common_lisp.rb]: On every build.

File: lib/travis/build/script/common_lisp.rb (hunk 3 above)
Comment (sionescu) [inline lib/travis/build/script/common_lisp.rb]: Not on update.

File: lib/travis/build/script/common_lisp.rb (hunk 4 above)
Comment (sionescu) [inline lib/travis/build/script/common_lisp.rb]: I think nowadays it works without.

File: lib/travis/build/script/common_lisp.rb (hunk 1 above)
Comment (sionescu) [inline lib/travis/build/script/common_lisp.rb]: Good point. Removed until we find a good default.

**Comment by luismbo** [comment]
Will this Common Lisp support documented somehow in the official Travis documentation?

**Comment by sionescu** [comment]
Description updated with the names.

**Comment by BanzaiMan** [comment]
Please add the maintainers' names to the code itself.

**Comment by sionescu** [comment]
Done.

**Comment by BanzaiMan** [comment]
Sorry for not being more explicit, but the maintainers' names need to be visible to the users, so that when they report problems on GitHub they know whom to cc. (Though often they forget this.)

**Comment by sionescu** [comment]
Done

File: lib/travis/build/script/common_lisp.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        ROS_URL = "https://raw.githubusercontent.com/roswell/roswell/release/scripts/install-for-ci.sh"
+
+        SYSTEM_MISSING = %Q[\
+No test script provided.
+Please either override the script: key
+]
+
+        def configure
+          super
+          sh.echo "If you encounter any issues with Common Lisp support, please Cc @tmccombs, @luismbo, @snmsts or @sionescu."
```
Comment (BanzaiMan) [inline lib/travis/build/script/common_lisp.rb]: I know the verbiage is not completely consistent, but could you take a look at other community-supported languages such as https://travis-ci.org/Vauxoo/runbot_branch_remote_name_grp_feature2/jobs/110692774#L1399 to mention GitHub Issues link?

Thanks.

**Comment by sionescu** [comment]
How about now ? Btw, this should probably go into the official documentation.

**Comment by BanzaiMan** [comment]
Let us test this on staging (https://staging.travis-ci.org) this Friday (Feb. 26 HST). Thanks.

**Comment by sionescu** [comment]
How will testing happen ? I'll be gone during the weekend so I probably won't be around to answer questions.

**Comment by BanzaiMan** [comment]
To test, I will deploy the PR, and you run the tests on staging. If you are not around, then we will have to figure out some other ways to synchronize the testing.

**Comment by snmsts** [comment]
Then I will test something. HST... Hawaii?

**Comment by luismbo** [comment]
I will try to be around for testing too.

**Comment by BanzaiMan** [comment]
Yes. Hawaii Standard Time. I'll update this thread when it is deployed to staging. Thanks.

**Comment by BanzaiMan** [comment]
This has been deployed to staging. Please test it.

**Comment by luismbo** [comment]
I won't be able to do any testing until tomorrow. @snmsts any luck?

**Comment by snmsts** [comment]
https://staging.travis-ci.org/snmsts/quri/builds/491735 do I have luck?
I need more working on it.

**Comment by luismbo** [comment]
@snmsts that doesn't look very lucky, no. :-)

**Comment by snmsts** [comment]
it's hard to test if the process are stop at anounce part... :-)

@BanzaiMan 

Can we try it again? with at least fe36998 applied.

**Comment by BanzaiMan** [comment]
Sorry about the delay. We'll try again on Friday.

**Comment by snmsts** [comment]
Thank you.

**Comment by BanzaiMan** [comment]
@9ab98e3 has been deployed to staging.

**Comment by BanzaiMan** [comment]
Did anyone test this last week? I deployed it to staging once again, but I might take it down for testing other things over the weekend.

**Comment by snmsts** [comment]
I'm sorry for not being update.
I did some test.

https://staging.travis-ci.org/snmsts/quri/jobs/492013

the result was something wired.It should be debug.

**Pull request closed without merging**